tests/test_devicetree_lexer.py

```diff
@@ -0,0 +1,25 @@
+"""
+    Devicetree Lexer Tests
+    ~~~~~~~~~~~~~~~~~~~~~~
+
+    :copyright: Copyright 2006-2021 by the Pygments team, see AUTHORS.
+    :license: BSD, see LICENSE for details.
+"""
+
+import pytest
+
+from pygments.lexers.devicetree import DevicetreeLexer
+from pygments.token import Token
+
+
+@pytest.fixture(scope='module')
+def lexer():
+    yield DevicetreeLexer()
+
+
+def test_fragment_out_of_root_node(lexer):
+    """Validate that a devicetree fragment out of a root node is parsed correctly."""
+
+    fragment = 'nodelabel: node@0 { foo = "bar"; };'
```

Comment:
Just confirming, but these fragments all work as expected (i.e. are valid)  with this PR, right?

1. `nodelabel: node { foo = "bar"; };`
2. `nodelabel1: nodelabel2: node@0 { foo = "bar"; };`
3. `nodelabel1: nodelabel2: node { foo = "bar"; };`

Reply:
good spot, multi label is not parsed correctly.

Reply:
actually, `nodelabel1: nodelabel2: node { foo = "bar"; };` is parsed correctly even with current implementation

Reply:
current lexer doesn't seem to play well when we have the combination of `label + node + @ + addr` out of root node.

Reply:
Makes sense since that's only valid in a snippet.

Reply:
latest changes should fix all the cases mentioned above.